engine/src/execution_graph/logic_controllers/CacheData.cpp

```diff
@@ -307,15 +307,13 @@ std::unique_ptr<ral::frame::BlazingTable> ConcatCacheData::decache() {
 		return _cache_datas[0]->decache();
 	}
 
-	std::vector<std::unique_ptr<ral::frame::BlazingTable>> tables_holder;
-	std::vector<ral::frame::BlazingTableView> table_views;
+	std::vector<std::unique_ptr<ral::frame::BlazingTable>> tables;
 	for (auto && cache_data : _cache_datas){
-		tables_holder.push_back(cache_data->decache());
-		table_views.push_back(tables_holder.back()->toBlazingTableView());
-
-		RAL_EXPECTS(!ral::utilities::checkIfConcatenatingStringsWillOverflow(table_views), "Concatenating tables will overflow");
+		tables.push_back(cache_data->decache());
+		
+		RAL_EXPECTS(!ral::utilities::checkIfConcatenatingStringsWillOverflow(tables), "Concatenating tables will overflow");
 	}
-	return ral::utilities::concatTables(table_views);
+	return ral::utilities::concatTables(std::move(tables));
```

Comment:
We used to only have one API for concatTables. One where it took in views. This made sense for the first uses of that function, but then later we needed to concat Tables and we were just created some temporary vectors of views just to be able to call this function API. So instead I encapsulated that into a different version of the same function and refactored some of the code to use the new API so that the code was more succinct